modules/lang-painless/src/main/java/org/elasticsearch/painless/Definition.java

```diff
@@ -202,16 +203,27 @@ public MethodType getMethodType() {
 
         public void write(MethodWriter writer) {
             final org.objectweb.asm.Type type;
+            final Class<?> clazz;
             if (augmentation != null) {
                 assert java.lang.reflect.Modifier.isStatic(modifiers);
+                clazz = augmentation;
                 type = org.objectweb.asm.Type.getType(augmentation);
             } else {
+                clazz = owner.clazz;
                 type = owner.type;
             }
 
             if (java.lang.reflect.Modifier.isStatic(modifiers)) {
-                writer.invokeStatic(type, method);
-            } else if (java.lang.reflect.Modifier.isInterface(owner.clazz.getModifiers())) {
+                // special case for interfaces where the interface function needs to be set to true
```

Comment:
I think this'd be more clear if you said something like "invokeStatic assumes that it is not invoking a method on an interface."

Reply:
Agreed.  Cleaned up a bit.